src/main/groovy/com/vherasymenko/avro/encoder/core/CourseInstallEncoderService.groovy

```diff
@@ -0,0 +1,78 @@
+package com.vherasymenko.avro.encoder.core
+
+import com.vherasymenko.avro.encoder.outbound.MessageProducer
+import event.course_install.Course
+import event.course_install.CourseInstall
+import event.course_install.Lesson
+import event.course_install.Unit
+import groovy.json.JsonSlurper
+
+/**
+ * Service that encodes course install document.
+ */
+class CourseInstallEncoderService implements CourseInstallEncoderPort {
+
+    /**
+     * The gateway to the messaging system.
+     */
+    private final MessageProducer producer
+
+    /**
+     * The avro json encoder.
+     */
+    private final AvroEncoderPort encoder
+
+    CourseInstallEncoderService( MessageProducer aProducer, AvroEncoderPort anEncoder ) {
+        producer = aProducer
+        encoder = anEncoder
+    }
+
+    @Override
+    void encodeEvent( String event ) {
+        def schema = CourseInstall.classSchema
+
+        def jsonParser = new JsonSlurper()
+        def jsonData = jsonParser.parseText( event )
+
+        // map the request data to the schema generated avro object
+        def avroData = new CourseInstall()
+        def course = new Course()
+        course.setUuid( jsonData.course.uuid )
```

Comment:
Too bad we have to do the mapping by hand.  This is a likely place for subtle errors.